<?xml version="1.0"?>
<launch>
  <!-- Standalone example launch file for 3DM-GX5-25 -->

  <!-- For setting debug level to debug -->
  <!--
  <env name="ROSCONSOLE_CONFIG_FILE"
       value="$(find microstrain_3dm_gx5_45)/config/custom_rosconsole.conf"/>
  -->

  <!-- Microstain sensor node -->
  <node name="microstrain_3dm_gx5_25_node"
  pkg="microstrain_3dm_gx5_45"
  type="microstrain_3dm_gx5_25_node" output="screen">
    <param name="port" value="/dev/ttyACM0" type="str" />
    <param name="baudrate" value="115200" type="int" />

    <param name="device_setup" value="true" type="bool" />
    <!-- General Settings -->
    <param name="readback_settings" value="true" type="bool" />
    <param name="save_settings" value="false" type="bool" />
    <param name="auto_init" value="true" type="bool" />
    <!-- AHRS Settings -->
    <param name="publish_imu" value="false" type="bool" />
    <param name="imu_raw_rate" value="50" type="int" />
    <param name="imu_frame_id" value="imu" type="str" />
    <param name="imu_raw_topic" value="imu_raw/data" type="str" />
    <!-- Setting manual declination is not yet implemented/testes -->
    <!-- Declination source 1=None, 2=magnetic, 3=manual -->
    <param name="declination_source" value="2" type="int" />
    <param name="declination" value="0.23" type="double" />

    <param name="publish_imu_filltered" value="true" type="bool" />
    <param name="imu_filtered_rate" value="50" type="int" />
    <param name="imu_filtered_topic" value="imu_filtered/data" type="str" />
    <!-- <param name="dynamics_mode" value="1" type="int" /> -->
  </node>

  <!-- Diagnostics -->
  <!--
  <node pkg="rqt_topic" type="rqt_topic" name="rqt_topic"/>
  -->
  <!--<node pkg="rqt_plot" type="rqt_plot" name="pid_setpoints"
      args="/yaw_pid_debug/Setpoint /vel_pid_debug/Setpoint"/>-->

</launch>
